### PR TITLE
output translator

### PR DESCRIPTION
**Pull request by benflexcompute**

### CODE REVIEW

File: flow360/component/simulation/translator/utils.py [hunk 1]
```diff
@@ -136,29 +132,70 @@ def remove_units_in_dict(input_dict):
     return input_dict
 
 
+def is_terminal_instance(obj, cls):
```
Comment (yifan-flex) [inline flow360/component/simulation/translator/utils.py:135]: Can you rename to is_exact_instance or is_direct_instance

File: flow360/component/simulation/translator/utils.py [hunk 2]
```diff
@@ -136,29 +132,70 @@ def remove_units_in_dict(input_dict):
...
+                    # We only look for empty entities instances
+                    # Note: This poses requirement that entity list has to be under attribute name 'entities'
+                    continue
+
+                # Route 2: Allowed to look into non-empty-entity instances
+                default_value = obj.model_fields[attribute_name].default
+                field_value = getattr(obj, attribute_name)
+                is_last_item = _is_last_of_type(obj_list, obj)
+
+                if (field_value == default_value) and (is_last_item is False):
```
Comment (yifan-flex) [inline flow360/component/simulation/translator/utils.py:191]: What's the current design of validation if two objects have different values of a global entry? Shouldn't this fail validation?

File: flow360/component/types.py [hunk 3]
```diff
@@ -98,9 +98,9 @@ def validate(cls, vector):
...
             return None
         vector = super().validate(vector)
-        vector_norm = 0.0
+        vector_norm = np.float64(0.0)
         for element in vector:
-            vector_norm += element * element
-        vector_norm = np.sqrt(vector_norm)
-        normalized_vector = tuple(e / vector_norm for e in vector)
+            vector_norm += np.float64(element) * np.float64(element)
+        vector_norm = np.sqrt(vector_norm, dtype=np.float64)
```
Comment (yifan-flex) [inline flow360/component/types.py:104]: Can you use np.linalg.norm

File: flow360/component/simulation/outputs/outputs.py [hunk 4]
```diff
@@ -129,7 +129,7 @@ class SliceOutput(_AnimationAndFileFormatSettings):
     """Slice output settings."""
 
     name: Optional[str] = pd.Field(None)
-    entities: UniqueItemList[Slice] = pd.Field(alias="slices")
+    entities: Optional[UniqueItemList[Slice]] = pd.Field(None, alias="slices")
```
[the remark this answers is not in the document]
Comment (benflexcompute) [reply flow360/component/simulation/outputs/outputs.py:132]: To support global setting for shared output fields.

File: flow360/component/simulation/translator/utils.py (hunk 2 above)
Comment (benflexcompute) [reply flow360/component/simulation/translator/utils.py:191]: To be added in solver capability validators

**Review by yifan-flex — APPROVED** [review]

**Pull request merged**